Guided Transfer: Redirect to wp-admin flow

jordwest opened the pull request:
This makes the Guided Transfer button visible regardless of the "manage/export/guided-transfer" feature flag, but redirects to the existing wp-admin flow while the Calypso flow is under development. Note that the wp-admin flow actually redirects to Calypso for payment, as you can see below:

![legacygt](https://cloud.githubusercontent.com/assets/416133/16032454/41058fb2-3249-11e6-95fe-6f6db7de54cf.gif)

Before this is ready we should:
 - [x] Ensure the legacy flow works correctly from start to finish with this redirect in place.
 - [x] Implement the blurb design to give the user context (#6048):

<img width="536" alt="screen shot 2016-06-14 at 4 03 42 pm" src="https://cloud.githubusercontent.com/assets/416133/16032496/96d30e9c-3249-11e6-9863-4fb9eab07a99.png">

dllh commented:
How can I test the legacy flow? I think I'm in a dev environment no matter what when I go to calypso.live, so I only get the flow that renders a blank screen for GT. I guess I could run it locally and disable in the dev environment to get the legacy flow. Is there a simpler way?

jordwest commented:
Give this link a go - it's just a branch from this one but with the guided transfer feature flag disabled in development:

https://calypso.live/?branch=add/guided-transfer/redirect-button-2

dllh commented:
Ah, thanks! I tested and it worked as expected when I purchased. The code looks fine to me too, but maybe get a quick sanity check by @omarjackman before we merge.

omarjackman commented on client/my-sites/exporter/index.jsx:
```diff
@@ -36,7 +35,6 @@ function mapStateToProps( state ) {
 		downloadURL: state.siteSettings.exporter.downloadURL,
 		didComplete: getExportingState( state, siteId ) === States.COMPLETE,
 		didFail: getExportingState( state, siteId ) === States.FAILED,
-		showGuidedTransferOptions: config.isEnabled( 'manage/export/guided-transfer' ),
```
I think we should keep the config stuff here and not move it into the component. Maybe what we need to do here is change the name from `showGuidedTransferOptions` to `isCalypsoGuidedTransferEnabled` or something that indicates it will use the wp-admin flow instead of the calypso flow.

jordwest replied:
I'd rather avoid having it in `mapStateToProps` as it's not derived from the Redux store state. While config values aren't going to change throughout the lifecycle of the app, it can be an issue putting non-Redux-state dependent props in `mapStateToProps` as it confuses Redux's update logic. If `state` doesn't change, Redux won't recompute `mapStateToProps`. 

Normally if there are non-Redux-state dependent props which need to be combined with Redux state, they can be passed in at the parent component level then accessed in `mapStateToProps` via the [`ownProps` argument](https://github.com/reactjs/react-redux/blob/master/docs/api.md#arguments). But in this case I don't think we need to make this a prop at all. With this PR the top level exporter component no longer needs to know whether the feature is enabled or not; the Guided Transfer card is shown regardless. Instead, we can just access it where it's needed, inside the component when the button is clicked. Once guided transfer is fully launched and running without issues, the goal is to remove the feature flag check anyway.

omarjackman commented:
@jordwest I left a comment about keeping the `config` logic out of the component. I suggest we just pass it down as a prop and give it a more descriptive name since it's meaning has changed slightly.

omarjackman commented:
@jordwest will we be enabling this config option in the staging environment in another PR?

jordwest commented:
> will we be enabling this config option in the staging environment in another PR?

Once Guided Transfer is finished and tested, yes 😃 

This PR is really just for the period until then; once the feature flag is enabled we can remove the redirect logic introduced here.

omarjackman commented:
LGTM 🚢

omarjackman commented on client/my-sites/exporter/guided-transfer-options.jsx:
```diff
@@ -18,7 +19,13 @@ export default React.createClass( {
+			page( `/settings/export/${ siteSlug }/guided` );
+		} else {
+			// Redirect to legacy guided transfer
+			window.location = `https://${ siteSlug }/wp-admin/paid-upgrades.php?product=40&view=purchase&source=export`;
```
@jordwest I think we should use the site url here instead of `siteSlug` by getting it from the store and passing it in as a prop. Then we could have 

```js
window.location = siteUrl + '/wp-admin/paid-upgrades.php?product=40&view=purchase&source=export';
```

omarjackman replied:
Also I'm not sure about the `product=40` part. If thats stored somewhere as a constant or anywhere else I think we could use it by name. If not then its probably not a big deal.

dllh replied:
There is a constant for it on the php side, but we'll be getting rid of this code anyway as soon as the new GT flow is ready, so I don't think it's worth worrying about too much.

jordwest replied:
> I think we should use the site url here instead of siteSlug

Ah, good point. I'll update.

> Also I'm not sure about the product=40 part.

Yeah that does feel a bit icky, but once #6143 and #6259 lands we can fetch the product id (and price) from the store.

omarjackman commented:
@jordwest I left another comment about the wp-admin link. Once thats taken care of feel free to :ship:

jordwest commented:
Thanks for the review @omarjackman 

I've added 7598d54 to use `site.URL` instead of `siteSlug`, and rebased. I'll wait until we launch Export next week to sort out any issues before merging this one

jordwest commented:
Closing this for now as it's out of date, plus we're close enough to launching the complete GT flow in Calypso.
